sklearn/utils/validation.py

```diff
@@ -982,3 +982,121 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
 
     if max_val is not None and x > max_val:
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
+
+
+def check_kernel_eigenvalues(lambdas, warn_on_zeros=False):
+    """Checks that the provided array of kernel eigenvalues for numerical or
+    conditioning issues and returns a fixed validated version.
+
+    It checks:
+
+     - that there are no significant imaginary parts in eigenvalues (more than
```

Comment:
Please use hanging indentation for restructured text lists

Reply:
I think I got what you mean: I remove the space before the dash and indent the text within each bullet as follows:

```python
    """Checks kernel eigenvalues for conditioning issues.

    Checks the provided array of kernel eigenvalues for numerical or
    conditioning issues and returns a fixed validated version.

    It checks:

    - that there are no significant imaginary parts in eigenvalues (more than
      1e-5 times the maximum real part). If this check fails, it raises a
      ValueError. Otherwise all non-significant imaginary parts that may remain
      are removed.

    - that eigenvalues are not all negative. If this check fails, it raises a
      ValueError

    ...
```